R/profileApp.R

```diff
@@ -1,53 +1,85 @@
 #' Safety Profile App
 #'
-#' @param dfAE AE Data
-#' @param dfDemog demog data
-#' @param settings safetyGraphics settings
+#' @param data `list` Named list of data domains
+#' @param settings `list` Named list of data mappings
+#' @param standard `character` Name of data standard
+#' @param runNow `logical` Run app?
 #'
-#' @import shiny
+#' @return `shiny.appobj` Shiny app
+#'
+#' @importFrom shiny shinyApp callModule
 #'
 #' @export
 
 profileApp <- function(
-    data = list(
-        aes=safetyData::adam_adae,
-        dm = safetyData::adam_adsl,
-        labs=safetyData::adam_adlbc
-    ),
-    settings=NULL,
-    runNow=TRUE
-){
-
-    ## create default settings when settings is not defined by default
-    if(is.null(settings)){
-        settings<-list(
-            labs=list(id_col="USUBJID"),
-            aes=list(id_col="USUBJID", siteid_col="SITEID", trarm_col="TRTA",
-                     bodsys_col="AEBODSYS", term_col = 'AEDECOD',
-                     aeterm_col="AETERM", severity_col="AESEV",
-                     stdy_col="ASTDY", endy_col="AENDY"),
-            dm=list(id_col="USUBJID", treatment_col="ARM")
+    data = NULL,
+    settings = NULL,
+    standard = 'adam',
+    runNow = TRUE
+) {
+    stopifnot(
+        '[ standard ] must be a character-classed variable.' =
+            class(standard) == 'character',
+        '[ standard ] must be one of "sdtm", "adam", "custom".' =
+            tolower(standard) %in% c('sdtm', 'adam', 'custom')
+    )
+
+    standard <- tolower(standard)
+
+    ## Create list of default data when undefined.
+    if (is.null(data))
+        data <- getDefaultData(standard)
+
+    ## Create list of default settings when undefined.
+    if (is.null(settings))
+        settings <- getDefaultSettings(standard)
+
+    ## Calculate study timing.
+    if (standard == 'sdtm') {
+        data$aes <- getTiming(
+            params = list(
+                data = data,
+                settings = settings,
+                standard = standard
+            ),
+            'aes',
+            domainStartDate = settings$aes$stdt_col,
+            domainEndDate = settings$aes$endt_col
+        )
+        settings$aes$stdy_col <- paste0(settings$aes$stdt_col, '_dy')
+        settings$aes$endy_col <- paste0(settings$aes$endt_col, '_dy')
+
+        data$labs <- getTiming(
+            params = list(
+                data = data,
+                settings = settings,
+                standard = standard
+            ),
+            'labs',
+            domainDate = settings$labs$dt_col
         )
+        settings$labs$dy_col <- paste0(settings$labs$dt_col, '_dy')
```

Comment:
I think we should treat this more like a utility function. Let's move this to a separate function (`getSDTMTiming`) and have the user call it before starting the app. That way we can use the day variable in other safetyGraphics modules as well if needed. Something like: 

```
sdtm <- getDefaultData(standard="sdtm") %>% getSDTMTiming
safetyGraphicsApp(data=sdtm)
# or profileApp(data=sdtm)
```